### PR TITLE
Same as 110, but works.

### PR DESCRIPTION
**Pull request by raynasr**
Up to date with the latest master branch.

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -1,7 +1,6 @@
-<?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="ExternalStorageConfigurationManager" enabled="true" />
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_11" default="true" project-jdk-name="Android Studio default JDK" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_17" default="true" project-jdk-name="Android Studio default JDK" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/build/classes" />
   </component>
   <component name="ProjectType">
```
Comment (edouardmichelin) [inline .idea/misc.xml]: Please remove from the commit

File: app/src/main/java/com/studhub/app/presentation/listing/add/CreateListingScreen.kt [hunk 2]
```diff
@@ -169,14 +170,52 @@ fun ListingForm(
...
 
 
-    BasicFilledButton(
+    val priceValidationResult = validatePrice(price.value)
+
+    if (priceValidationResult != PriceValidationResult.VALID) {
+        Text(
+            text = when (priceValidationResult) {
+                PriceValidationResult.NEGATIVE -> "Please enter a non-negative price"
+                PriceValidationResult.NON_NUMERIC -> "Please enter a valid price"
```
Comment (edouardmichelin) [inline app/src/main/java/com/studhub/app/presentation/listing/add/CreateListingScreen.kt:179]: You should make use of string resources:

# 1. Declare and assign your string to some key in `res/values/strings_listings.xml`
For instance add
```xml
<?xml version="1.0" encoding="utf-8"?>
<resources>
    ...
    <string name="listing_add_form_validation_neg_price">Please enter a non-negative price</string>
</resources>
```
# 2. Make use of the string resrouce in your composable
```kt
PriceValidationResult.NEGATIVE -> stringResource(R.string.listing_add_form_validation_neg_price)
```

File: app/src/main/java/com/studhub/app/presentation/listing/add/CreateListingScreen.kt [hunk 3]
```diff
@@ -169,14 +170,52 @@ fun ListingForm(
...
+    if (price.isEmpty()) {
+        return PriceValidationResult.EMPTY
+    }
+    val parsedPrice = price.toDoubleOrNull()
+    return when {
+        parsedPrice == null -> PriceValidationResult.NON_NUMERIC
+        parsedPrice < 0 -> PriceValidationResult.NEGATIVE
+        else -> PriceValidationResult.VALID
+    }
 }
```
Comment (edouardmichelin) [inline app/src/main/java/com/studhub/app/presentation/listing/add/CreateListingScreen.kt:219]: I think we could use validators in more generic files like in `core/utils/FormValidator.kt`

File: app/src/main/java/com/studhub/app/presentation/profile/components/EditProfileForm.kt [hunk 4]
```diff
@@ -44,11 +47,38 @@ fun EditProfileForm(
             BasicTextField(stringResource(R.string.profile_edit_form_label_lastname), lastName)
             BasicTextField(stringResource(R.string.profile_edit_form_label_username), userName)
             NumericTextField(stringResource(R.string.profile_edit_form_label_phone), phoneNumber)
+
+
+            // Display an error message if the phone number is not valid
+            val isPhoneNumberValid = isValidSwissPhoneNumber(phoneNumber.value)
+            if (!isPhoneNumberValid) {
+                Text(
+                    text = stringResource(R.string.profile_edit_form_error_invalid_phone),
```
Comment (edouardmichelin) [inline app/src/main/java/com/studhub/app/presentation/profile/components/EditProfileForm.kt:56]: Nice!

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -9,5 +9,5 @@
     <string name="title_activity_create_listing">CreateListingActivity</string>
     <string name="title_activity_browse">BrowseActivity</string>
     <string name="title_activity_detailed_listing_view">DetailedListingView</string>
-
+    <string name="profile_edit_form_error_invalid_phone">Please enter a valid Swiss phone number</string>
```
Comment (edouardmichelin) [inline app/src/main/res/values/strings.xml:12]: This one should be moved to `res/values/strings_profile.xml`

**Review by edouardmichelin — CHANGES_REQUESTED** [review]

File: .idea/misc.xml (hunk 1 above)
Comment (raynasr) [reply .idea/misc.xml]: Reverted!

File: app/src/main/java/com/studhub/app/presentation/listing/add/CreateListingScreen.kt (hunk 2 above)
Comment (raynasr) [reply app/src/main/java/com/studhub/app/presentation/listing/add/CreateListingScreen.kt:179]: Done!

File: app/src/main/res/values/strings.xml (hunk 5 above)
Comment (raynasr) [reply app/src/main/res/values/strings.xml:12]: Done

**Review by edouardmichelin — APPROVED** [review]
It's a nice thing to have general usage validators in the `core.utils` package. Maybe in the future we'll put all validators together under a bigger "validator" class but this will do the job for now.

Good job overall
LGTM

**Pull request merged**